internal/handlers/common/update_array_operators.go

```diff
@@ -315,3 +315,100 @@ func processAddToSetArrayUpdateExpression(doc, update *types.Document) (bool, er
 
 	return changed, nil
 }
+
+// processPullAllArrayUpdateExpression changes document according to $pullAll array update operator.
+func processPullAllArrayUpdateExpression(doc, update *types.Document) (bool, error) {
+	var changed bool
+
+	iter := update.Iterator()
+
+	for {
+		key, pullAllValueRaw, err := iter.Next()
+		if err != nil {
+			if errors.Is(err, iterator.ErrIteratorDone) {
+				break
+			}
+
+			return false, lazyerrors.Error(err)
+		}
+
+		path, err := types.NewPathFromString(key)
+		if err != nil {
+			return false, lazyerrors.Error(err)
+		}
+
+		// If the path does not exist, create a new array and set it.
+		if !doc.HasByPath(path) {
+			if err = doc.SetByPath(path, types.MakeArray(1)); err != nil {
+				return false, commonerrors.NewWriteErrorMsg(
+					commonerrors.ErrUnsuitableValueType,
+					err.Error(),
+				)
```

Comment:
Can we cover is by test? Maybe something where we try to set a value to a non-suitable field? Like `update: bson.D{{"$pullAll", bson.D{{"v.pampam", bson.A{int32(42)}}}}},` where `v` is a scalar, so it's not possible to set `pampam` field

Reply:
Added.